Added Social Media Icons to Introduction Page

Kgkunal opened the pull request:
**What type of PR is this?**
> /kind enhancement

**What this PR does / why we need it**:
This PR enhances the Introduction page of the SODA Foundation's documentation by adding social media icons (Twitter, Slack, LinkedIn, etc.) alongside the existing textual links. This improvement aims to provide a more visually appealing and user-friendly experience, allowing users to easily access the foundation's social media channels.

**Which issue(s) this PR fixes**:
Fixes #83 

**Test Report Added?**:
> /kind TESTED

**Test Report**:
Tested the changes locally by rendering the markdown file with the added icons and verified that the icons display correctly and link to the respective social media channels.

**Test Preview**:

![image_2023-09-17_012803383](https://github.com/sodafoundation/documentation/assets/76809868/d38d23f5-bb50-4979-a540-044ff50e0b5f)

**Special notes for your reviewer**:
No special notes at this time.

Kgkunal commented:
Hi @rajat-soda and @anvithks,

I'd like to kindly request your review of this PR. It enhances the Introduction page of our documentation by adding social media icons for improved user experience. Your feedback and insights would be greatly appreciated.

Thank you in advance for taking the time to review.....

anvithks commented:
@Kgkunal This is how the page looks with your changes
![image](https://github.com/sodafoundation/documentation/assets/19162717/50833a45-c8ca-4523-b9fa-6ff41b541103)

You can check out the same in the deploy preview [here](https://deploy-preview-311--soda-docs.netlify.app/introduction/)

The alignment seems to be off. The logo size is very small as well. Could you try to fix these ?

Kgkunal commented:
Hi @anvithks,

Thank you for your feedback and for pointing out the alignment and logo size issues. I apologize for any inconvenience this may have caused.

I've taken your feedback into account, and I understand that the differences you observed between the VS Code preview and the deploy preview are due to the way the VS Code extension renders Markdown files versus how they appear in the deployed environment. VS Code may have some variations in rendering, and the deploy preview provides a more accurate representation of how the changes will appear on the actual website.

I appreciate your understanding and patience as we work to align the changes with the desired standards. I'll prioritize addressing the alignment and logo size issues to ensure they meet the requirements for the deployed site.

Once the adjustments are made, I will update the PR for your review.

Kgkunal commented:
@anvithks please guide me how can I check deployment preview of the page in my system. In my VSCode setup.. preview is different.

anvithks commented:
> @anvithks please guide me how can I check deployment preview of the page in my system. In my VSCode setup.. preview is different.

Hey @Kgkunal to setup the documentation on your local environment you can run the `./install_local.sh` script and follow the on screen instructions. 

This script works on Ubuntu but you can setup the documentation on other OS by installing the pre-requistes.
- Hugo
- Learn theme for Hugo. 

All of this is taken care by the `install_local.sh` script

Kgkunal commented:
@anvithks hello ,

I installed Hugo as well as learn theme , but whenever I run : install_local.sh command  , i get this error

Kgkunal commented:
@anvithks  hello , 
I've resolved the alignment and icon size issues, and I've tested the changes locally. Everything looks good on my end, and the social media icons are rendering correctly. Please review and merge the PR at your earliest convenience.

Thank you.....

Preview : 
![image_2023-09-20_043113700](https://github.com/sodafoundation/documentation/assets/76809868/14b0805c-d2c2-49ea-8be5-2c285535703d)

anvithks commented on content/introduction/_index.md:
```diff
@@ -49,57 +53,60 @@ As mentioned, SODA Architecture is getting refined and optimized for different a
- - [SODA Twitter](https://twitter.com/sodafoundation)  
- - [SODA Mailing List](https://lists.sodafoundation.io)
- - [SODA LinkedIn](https://www.linkedin.com/company/sodafoundation/)
+<ul style="list-style: none; ">
```
Avoid using inline styles.
You can use classes and  add custom styles to this file `documentation/static/css/theme-opensds.css`
This is the custom theme file that is being used for the website.

Kgkunal replied:
Thanks for the feedback! I'll make the changes you suggested to the theme-opensds.css file .

anvithks commented on go.mod:
```diff
@@ -1,3 +1,5 @@
 module github.com/sodafoundation/documentation
```
No need to check in this file.

anvithks commented on go.sum:
```diff
@@ -0,0 +1,2 @@
+github.com/soda-cdm/documentation v0.0.0-20230525043849-2b985763d78d h1:3xzxcT2yHfhVJWOvnVstPMU9lx3yzZ24RAojwVvHC6I=
```
No need to check in this file

anvithks replied:
Seems like this file is still checked in. 
Since you had it pushed it to the fork already you will have to delete the file and push the changes again.

anvithks commented on content/introduction/_index.md:
```diff
@@ -6,16 +6,18 @@ disableToc: false
 ---
 
 ## Introduction
+
```
Just trying to understand why the newline here and spaces throughout the document have been added. 
Was it done by a prettifier plugin?

Kgkunal replied:
Yes, the newlines and spaces were added by a prettifier plugin automatically, and I wasn't aware of it. I'll ensure it's disabled to maintain the original formatting.

anvithks reviewed (changes requested):
Thanks @Kgkunal.
Some minor review comments. 
Please check.

Kgkunal commented:
Hi @anvithks ,

I've made the requested changes in response to your feedback. Here's a summary of the updates:

1. Removed inline styling for the social media links in the markdown file.
2. Added classes to the links for external stylesheet styling.
3. Removed the unnecessary `go.mod` file.
4. Addressed the issue with newline characters and spaces throughout the document, which were not intentional.

I've tested these changes locally, and they appear to work as expected. Please review the updates and let me know if there are any additional changes or improvements needed.

Thank you for your guidance and feedback!

skdwriting commented on content/introduction/_index.md:
```diff
@@ -49,57 +53,61 @@ As mentioned, SODA Architecture is getting refined and optimized for different a
- - [SODA Mailing List](https://lists.sodafoundation.io)
- - [SODA LinkedIn](https://www.linkedin.com/company/sodafoundation/)
+<ul class="social-list">
+  <li class="social-list-item"><a href="https://sodafoundation.io/the-foundation/join/"><i class="fas fa-globe"></i> SODA Website</a></li>
```
Please make this link points to - https://www.sodafoundation.io/

skdwriting commented on content/introduction/_index.md:
```diff
@@ -49,57 +53,61 @@ As mentioned, SODA Architecture is getting refined and optimized for different a
+  <li class="social-list-item"><a href="https://lists.sodafoundation.io"><i class="fas fa-envelope"></i> SODA Mailing List</a></li>
+  <li class="social-list-item"><a href="https://www.linkedin.com/company/sodafoundation/"><i class="fab fa-linkedin"></i> SODA LinkedIn</a></li>
+</ul>
+
```
We can also add a new link:
To know the introduction and how to contribute, you can refer [SODA Starter](https://bit.ly/soda-starter)

anvithks reviewed (changes requested):
@Kgkunal please remove the go.mod and go.sum file and check in again. Please check @skdwriting comments as well.

Kgkunal commented:
Hello @skdwriting and @anvithks ,

I hope you're doing well. I wanted to address the request to remove the go.mod and go.sum files in the pull request.

I attempted to remove these files as requested, and it resulted in all the checks failing.  if there's a different approach you'd like me to take, please let me know.....
